sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -691,6 +691,13 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                         DBSPType rightType = right.getType();
                         if (!rightType.is(DBSPTypeInteger.class))
                             throw new UnimplementedException("ROUND expects a constant second argument", node);
+
+                        // convert to int32
+                        DBSPTypeInteger rightInt = rightType.to(DBSPTypeInteger.class);
```

Comment:
This will throw for a DECIMAL type. Is there a test for that case?

Reply:
The right type cannot be a decimal.
Calcite gives an error preventing compilation. 

And even if that didn't occur, in the line above, we throw if the `rightType` isn't an integer. 

For `SELECT round(15.1, 1.0);` we get: 
```java
(no input file): Error in SQL statement
(no input file):1:26: error: Error in SQL statement: Cannot apply 'ROUND' to arguments of type 'ROUND(<DECIMAL(3, 1)>, <DECIMAL(2, 1)>)'. Supported form(s): 'ROUND(<NUMERIC>, <INTEGER>)'
    1|CREATE VIEW VV AS SELECT round(15.1, 1.0)
                               ^^^^^^^^^^^^^^^^


java.lang.RuntimeException: org.apache.calcite.runtime.CalciteContextException: From line 1, column 26 to line 1, column 41: Cannot apply 'ROUND' to arguments of type 'ROUND(<DECIMAL(3, 1)>, <DECIMAL(2, 1)>)'. Supported form(s): 'ROUND(<NUMERIC>, <INTEGER>)'

	at org.dbsp.sqlCompiler.compiler.DBSPCompiler.compileInternal(DBSPCompiler.java:289)
```